In [1]:
import pandas as pd
import json

In [2]:
ads_list = []

with open("scout_car/scout_car.json", 'r') as file:
    for ad in file:
        ads_list.append(json.loads(ad))
        
autos = pd.DataFrame(ads_list)

autos.head()



url make_model  \
0  https://www.autoscout24.com//offers/audi-a1-sp...    Audi A1   
1  https://www.autoscout24.com//offers/audi-a1-1-...    Audi A1   
2  https://www.autoscout24.com//offers/audi-a1-sp...    Audi A1   
3  https://www.autoscout24.com//offers/audi-a1-1-...    Audi A1   
4  https://www.autoscout24.com//offers/audi-a1-sp...    Audi A1   

                                   short_description body_type  price  \
0        Sportback 1.4 TDI S-tronic Xenon Navi Klima    Sedans  15770   
1                                     1.8 TFSI sport    Sedans  14500   
2  Sportback 1.6 TDI S tronic Einparkhilfe plus+m...    Sedans  14640   
3                            1.4 TDi Design S tronic    Sedans  14500   
4  Sportback 1.4 TDI S-Tronic S-Line Ext. admired...    Sedans  16790   

                vat         km registration         prev_owner    kW  ...  \
0    VAT deductible  56,013 km      01/2016  2 previous owners  None  ...   
1  Price negotiable  80,000 km      03/2017               None  None  ...   
2    VAT deductible  83,450 km      02/2016   1 previous owner  None  ...   
3              None  73,000 km      08/2016   1 previous owner  None  ...   
4              None  16,200 km      05/2016   1 previous owner  None  ...   

                                         description   Emission Label  \
0  [\n, Sicherheit:,  , Deaktivierung für Beifahr...              NaN   
1  [\nLangstreckenfahrzeug daher die hohe Kilomet...  [\n4 (Green)\n]   
2  [\n, Fahrzeug-Nummer: AM-95365,  , Ehem. UPE 2...  [\n4 (Green)\n]   
3  [\nAudi A1: , - 1e eigenaar , - Perfecte staat...              NaN   
4  [\n, Technik & Sicherheit:, Xenon plus, Klimaa...              NaN   

     Gears Country version Electricity consumption Last Service Date  \
0      NaN             NaN                     NaN               NaN   
1  [\n7\n]             NaN                     NaN               NaN   
2      NaN             NaN                     NaN               NaN   
3  [\n6\n]             NaN                     NaN               NaN   
4      NaN   [\nGermany\n]                     NaN               NaN   

  Other Fuel Types Availability Last Timing Belt Service Date Available from  
0              NaN          NaN                           NaN            NaN  
1              NaN          NaN                           NaN            NaN  
2              NaN          NaN                           NaN            NaN  
3              NaN          NaN                           NaN            NaN  
4              NaN          NaN                           NaN            NaN  

[5 rows x 54 columns]

In [3]:
autos.make_model.unique()

array(['Audi A1', 'Audi A2', 'Audi A3', 'Opel Astra', 'Opel Corsa',
       'Opel Insignia', 'Renault Clio', 'Renault Duster',
       'Renault Espace'], dtype=object)

In [4]:
autos.isnull().sum()

url                                  0
make_model                           0
short_description                   46
body_type                           60
price                                0
vat                               4513
km                                   0
registration                         0
prev_owner                        6828
kW                               15919
hp                                   0
Type                                 2
Previous Owners                   6640
Next Inspection                  12384
Inspection new                   11987
Warranty                          5420
Full Service                      7704
Non-smoking Vehicle               8742
null                                 0
Make                                 0
Model                                0
Offer Number                      3175
First Registration                1597
Body Color                         597
Paint Type                        5772
Body Color Original      

In [5]:
import warnings
warnings.filterwarnings('ignore')
warnings.warn("this will not show")
# autos.groupby(["make_model","Weight"])["url"].count()

autos.km.unique()

array(['56,013 km', '80,000 km', '83,450 km', ..., '2,864 km', '1,506 km',
       '57 km'], dtype=object)

In [31]:
autos.Type[:60]

0               [, Used, , Diesel (Particulate Filter)]
1                                  [, Used, , Gasoline]
2               [, Used, , Diesel (Particulate Filter)]
3               [, Used, , Diesel (Particulate Filter)]
4               [, Used, , Diesel (Particulate Filter)]
5               [, Used, , Diesel (Particulate Filter)]
6               [, Used, , Diesel (Particulate Filter)]
7               [, Used, , Diesel (Particulate Filter)]
8               [, Used, , Diesel (Particulate Filter)]
9                                  [, Used, , Gasoline]
10              [, Used, , Diesel (Particulate Filter)]
11                                 [, Used, , Super 95]
12                       [, Used, , Regular/Benzine 91]
13                                 [, Used, , Gasoline]
14                                 [, Used, , Super 95]
15    [, Employee's car, , Diesel (Particulate Filter)]
16                                   [, Used, , Diesel]
17              [, Used, , Diesel (Particulate F

In [7]:
autos.F

Index(['url', 'make_model', 'short_description', 'body_type', 'price', 'vat',
       'km', 'registration', 'prev_owner', 'kW', 'hp', 'Type',
       'Previous Owners', 'Next Inspection', 'Inspection new', 'Warranty',
       'Full Service', 'Non-smoking Vehicle', 'null', 'Make', 'Model',
       'Offer Number', 'First Registration', 'Body Color', 'Paint Type',
       'Body Color Original', 'Upholstery', 'Body', 'Nr. of Doors',
       'Nr. of Seats', 'Model Code', 'Gearing Type', 'Displacement',
       'Cylinders', 'Weight', 'Drive chain', 'Fuel', 'Consumption',
       'CO2 Emission', 'Emission Class', '\nComfort & Convenience\n',
       '\nEntertainment & Media\n', '\nExtras\n', '\nSafety & Security\n',
       'description', 'Emission Label', 'Gears', 'Country version',
       'Electricity consumption', 'Last Service Date', 'Other Fuel Types',
       'Availability', 'Last Timing Belt Service Date', 'Available from'],
      dtype='object')

#### the columns will be droped

- description
- null
- url
- Make
- Model
- kW
- Offer Number
- country version
- Electricity consumption          15782
- Last Service Date                15353
- Other Fuel Types                 15039
- Availability                     15284
- Last Timing Belt Service Date    15903
- Available from                   15647
- body color original 
- short description
- Type
- Full service
- Next Inspection                  12384
- Inspection new                   11987
- non-smoking vehicle
- first registration
- paint type
- body color original
- model code
#### the cloumns are important

*****************
##### p.s.
array(['Audi A1', 'Audi A2', 'Audi A3', 'Opel Astra', 'Opel Corsa',
       'Opel Insignia', 'Renault Clio', 'Renault Duster',
       'Renault Espace'], dtype=object)
*****************
- make model
- registration mm/yyyy
- km
- body type
- price
- Fuel
- cylinders
- hp
- CO2 Emission
- Gearing Type
- Displacement
- consumption ( combined can be taken)
- Gears
- \nComfort & Convenience\n          920 (number of features ?)
- \nEntertainment & Media\n         1374 (number of features ?)
- \nExtras\n                        2962 (number of features ?)
- Drive chain ?????
- Body                                60
- Nr. of Doors                       212
- Nr. of Seats                       977
- weight
- upholstery  (cloth or leder ?)
- Nr. of Doors                       212
- Nr. of Seats                       977
- body color
- vat (~ 2/3 of the cars are vat deductible, 2.6 % of the cars are price negotiable, use this informations by filling missing values)
- number of prewious owner (sample or choice)
  
   1. previous owner     8294
   2. previous owners     778
   3. previous owners      17
   4. previous owners       2
- warranty ???? %d%d months can be taken 


In [89]:
autos['Model Code'].value_counts()

TypeError: unhashable type: 'list'

Exception ignored in: 'pandas._libs.index.IndexEngine._call_map_locations'
Traceback (most recent call last):
  File "pandas\_libs\hashtable_class_helper.pxi", line 1652, in pandas._libs.hashtable.PyObjectHashTable.map_locations
TypeError: unhashable type: 'list'


[\n0035/BCB\n]    268
[\n0588/BNO\n]    245
[\n0588/BDB\n]    206
[\n0588/BHX\n]    188
[\n1844/ADJ\n]    175
                 ... 
[\n0035/BHX\n]      1
[\n0035/AWM\n]      1
[\n0035/BBT\n]      1
[\n0035/AND\n]      1
[\n0035/ASN\n]      1
Name: Model Code, Length: 232, dtype: int64

In [86]:
autos.registration

0        01/2016
1        03/2017
2        02/2016
3        08/2016
4        05/2016
          ...   
15914        -/-
15915    01/2019
15916    03/2019
15917    06/2019
15918    01/2019
Name: registration, Length: 15919, dtype: object